padl/utils.py

```diff
@@ -61,3 +63,122 @@ def __getattr__(self, attr):
+    or :meth:`padl.transforms.Transform.pd_postprocess`) and each level deeper moves recursively
+    inside the element that failed until the :class:`AtomicTransform` that got the Exception is
+    reached.
+    """
+    def __init__(self):
+        self.trans = None
+        self.args = None
+        self.default_msg = (
+            'Defined commands are: \n'
+            '   u(p): step up\n'
+            '   d(own): step down\n'
+            '   w(here am I?): show code position\n'
+            '   i(nput): show input here\n'
+            '   r(epeat): repeat here (will produce the same exception)\n'
+            '   t(ransform): displays the current transform\n'
+            '   h(elp): print help about the commands\n'
+            '   q(uit): quit'
+            '     -> this will store the input at this level in debug.args\n'
+            '        and the transform in debug.trans\n'
+        )
+
+    def __call__(self) -> None:
+        """
+        Call me in case of error.
+        User can give following input and expect response
+            u(p): step up\n'
+            d(own): step down\n'
+            w(here am I?): show code position\n'
+            i(nput): show input here\n'
+            r(epeat): repeat here (will produce the same exception)\n'
+            h(elp): print help about the commands\n
+            q(uit): quit'
+        """
+        pos = len(_pd_trace) - 1
+        print(self.default_msg + '\n' + _pd_trace[pos].transform_str)
+
+        while True:
+            try:
+                x = input('> ')
+            except IndexError:
+                continue
+            if x == 'd':
+                pos, msg = self._down_step(pos, _pd_trace)
+            elif x == 'u':
+                pos, msg = self._up_step(pos, _pd_trace)
+            elif x == 'q':
+                self.args = _pd_trace[pos].args
+                self.trans = _pd_trace[pos].transform
+                break
+            elif x == 'w':
+                msg = _pd_trace[pos].code_position
+            elif x == 'i':
+                msg = _pd_trace[pos].args
+            elif x == 'r':
+                self.args = _pd_trace[pos].args
+                self.trans = _pd_trace[pos].transform
+                # This 0 is because the last element carries a problem: when adding the last
+                # element to _pd_trace *Transform.pd_mode* has been already set up to None again.
+                self.repeat(_pd_trace[0].pd_mode, pos)
+            elif x == 'h' or x == 'help':
+                msg = self.default_msg
+            elif x == 't':
+                msg = _pd_trace[pos].transform_str
+            else:
+                i = _pd_trace[pos].args
+                try:
+                    code = compile(x, '', 'single')
+                    exec(code)
+                except Exception as err:
+                    print(err)
+                finally:
+                    continue
```

Comment:
It seems like `continue` is not allowed inside of a `finally` block for python 3.7. I think you can just remove the `finally` block.

```
E     File "/home/runner/work/padl/padl/padl/utils.py", line 147
E       continue
E       ^i
E   SyntaxError: 'continue' not supported inside 'finally' clause
```